File: apps/high-steak-mobile/lib/utils/google_maps.dart
import '../models/place.dart';

String googleMapsUrlForPlace(PlaceSummary place) {
  final lat = place.latitude.trim();
  final lng = place.longitude.trim();
  if (lat.isNotEmpty && lng.isNotEmpty && lat != '0' && lng != '0') {
    final query = Uri.encodeComponent('$lat,$lng');
    return 'https://www.google.com/maps/search/?api=1&query=$query';
  }
  final label = place.formattedAddress != null && place.formattedAddress!.isNotEmpty
      ? '${place.name}, ${place.formattedAddress}'
      : place.name;
  return 'https://www.google.com/maps/search/?api=1&query=${Uri.encodeComponent(label)}';
}

File: apps/high-steak-web/src/utils/googleMaps.ts
import type { PlaceSummary } from '../api/client'

function coordinateText(value: string | number | null | undefined): string {
  if (value == null) return ''
  return String(value).trim()
}

export function googleMapsUrlForPlace(place: PlaceSummary): string {
  const lat = coordinateText(place.latitude)
  const lng = coordinateText(place.longitude)
  if (lat && lng) {
    return `https://www.google.com/maps/search/?api=1&query=${encodeURIComponent(`${lat},${lng}`)}`
  }
  const label = place.formattedAddress
    ? `${place.name}, ${place.formattedAddress}`
    : place.name
  return `https://www.google.com/maps/search/?api=1&query=${encodeURIComponent(label)}`
}
